### PR TITLE
[ci] fix check-transaction-versions

### PR DESCRIPTION
**Pull request by alvicsam**
Allow fail the job, more info https://github.com/paritytech/polkadot/issues/6422
cc https://github.com/paritytech/ci_cd/issues/695

### CODE REVIEW

**Review by rcny — APPROVED** [review]

File: scripts/ci/gitlab/pipeline/build.yml [hunk 1]
```diff
@@ -53,7 +53,8 @@ check-transaction-versions:
     - npm install --ignore-scripts -g @polkadot/metadata-cmp
     # Set git config
     - git config remote.origin.url "https://github.com/paritytech/polkadot.git"
-    - git fetch origin release
+    - export LATEST_RELEASE=$(curl --silent "https://api.github.com/repos/paritytech/polkadot/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
```
Comment (joao-paulo-parity) [inline scripts/ci/gitlab/pipeline/build.yml:56]: use `jq -e -r '.tag_name'` to extract the tag name

also need to use the `GITHUB_TOKEN` in the `Authorization` header or your request will be rate-limited by the GitHub API

everything put together:

`    - export LATEST_RELEASE=$(curl -sSLf  -H "Authorization: token $GITHUB_TOKEN" "https://api.github.com/repos/paritytech/polkadot/releases/latest" | jq -e -r '.tag_name')`

**Review by joao-paulo-parity — APPROVED** [review]

**Comment by joao-paulo-parity** [comment]
@alvicsam I see that https://github.com/paritytech/ci_cd/issues/694 is linked in the description. Is it by accident? I don't notice any change related to that in this PR.

**Comment by alvicsam** [comment]
@joao-paulo-parity sorry, wrong number of PR

**Review by joao-paulo-parity — CHANGES_REQUESTED** [review]
Approved before by accident. I think https://github.com/paritytech/polkadot/pull/6425#discussion_r1046119264 should be dealt with before merge.

**Review by joao-paulo-parity — APPROVED** [review]

**Pull request merged**